code/modules/clothing/spacesuits/flightsuit.dm

```diff
@@ -0,0 +1,796 @@
+	stabilizer = 1
+
+/obj/item/device/flightpack/proc/disable_stabilizers()
+	if(wearer)
+		if(brake)
+			disable_airbrake()
+		wearer << "<span class='warning'>Your [src.name]'s stabilizers cut off, fins folding in and maneuvering thrusters shutting off!</span>"
+	stabilizer = 0
+
+/obj/item/device/flightpack/proc/activate_booster()
+	if(boost_charge < 5)
+		wearer << "<span class='warning'>Your [src.name] beeps an alert. Its boost capacitors are still charging!</span>"
+		return 0
+	wearer << "<span class='notice'>Flightpack: Boosters engaged!</span>"
+	wearer.visible_message("<span class='notice'>[wearer.name]'s flightpack engines flare in intensity as they are rocketed forward by the immense thrust!</span>")
+	boost = 1
+
+/obj/item/device/flightpack/proc/deactivate_booster()
+	wearer << "<span class='warning'>Flightpack: Boosters disengaged!</span>"
+	boost = 0
+
+/obj/item/device/flightpack/proc/enable_airbrake()
+	if(wearer)
+		if(!stabilizer)
+			enable_stabilizers()
+		wearer << "<span class='notice'>You enable your suit's airbrakes, slowing you down to a slow cruise.</span>"
+	brake = 1
+
+/obj/item/device/flightpack/proc/disable_airbrake()
+	if(wearer)
+		wearer << "<span class='notice'>You disable your suit's airbrakes, speeding up back to normal.</span>"
+	brake = 0
+
+/obj/item/device/flightpack/on_mob_move(dir, mob)
+	wearer_movement(dir)
+
+//MOB MOVEMENT STUFF----------------------------------------------------------------------------------------------------------------------------------------------
+
+/mob/proc/get_flightpack()
+	return
+
+/mob/living/carbon/get_flightpack()
+	var/obj/item/device/flightpack/F = back
+	if(istype(F))
+		return F
+
+/obj/item/device/flightpack/proc/allow_thrust()
+	return 1
+
+//FLIGHT SHOES FOR MOVEMENT DETECTION------------------------------------------------------------------------------------------------------------------------------
+
+/obj/item/clothing/shoes/flightshoes
+	name = "flight shoes"
+	desc = "A pair of specialized boots that contain stabilizers and sensors nessacary for flight gear to work" //Apparently you need these to detect mob movement.
+	icon_state = "flightshoes"
+	item_state = "flightshoes_mob"
+	var/obj/item/clothing/suit/space/hardsuit/flightsuit/suit = null
+	var/obj/item/device/flightpack/pack = null
+	var/mob/living/carbon/human/wearer = null
+	var/active = 0
+	resistance_flags = FIRE_PROOF
+
+/obj/item/clothing/shoes/flightshoes/Destroy()
+	if(suit)
+		suit.shoes = null
+	if(pack)
+		pack.shoes = null
+
+/obj/item/clothing/shoes/flightshoes/proc/toggle(toggle)
+	if(suit)
+		active = toggle
+		if(active)
+			src.flags |= NOSLIP
+		if(!active)
+			src.flags &= ~NOSLIP
+
+/obj/item/clothing/shoes/flightshoes/dropped(mob/wearer)
+	..()
+
+/obj/item/clothing/shoes/flightshoes/item_action_slot_check(slot)
+	if(slot == slot_shoes)
+		return 1
+
+/obj/item/clothing/shoes/flightshoes/equipped(mob/wearer, slot)
+	..()
```

Comment:
WIP stuff, putting it there for future use, I'l remove it if I don't find anything that needs it :)